rcl/src/rcl/arguments.c

```diff
@@ -267,6 +302,50 @@ rcl_parse_arguments(
         continue;
       }
 
+      // Attempt to parse argument as parameter override flag
+      if (strcmp(RCL_PARAM_FLAG, argv[i]) == 0 || strcmp(RCL_SHORT_PARAM_FLAG, argv[i]) == 0) {
+        if (i + 1 < argc) {
+          // Attempt to parse next argument as parameter override rule
+          if (RCL_RET_OK == _rcl_parse_param_rule(argv[i + 1], args_impl->parameter_overrides)) {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "param override rule : %s\n", argv[i + 1]);
+            i += 1;  // Skip flag here, for loop will skip rule.
+            continue;
+          } else {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
+              "Couldn't parse arg %d (%s) as parameter override rule. Error: %s", i + 1,
+              argv[i + 1], rcl_get_error_string().str);
+          }
+        } else {
+          RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
+            "Couldn't parse arg %d (%s) as parameter override flag. No rule found.\n",
+            i, argv[i]);
+        }
+        rcl_reset_error();
+      }
+
+      // Attempt to parse argument as remap rule flag
+      if (strcmp(RCL_REMAP_FLAG, argv[i]) == 0 || strcmp(RCL_SHORT_REMAP_FLAG, argv[i]) == 0) {
+        if (i + 1 < argc) {
+          // Attempt to parse next argument as remap rule
+          rcl_remap_t * rule = &(args_impl->remap_rules[args_impl->num_remap_rules]);
+          *rule = rcl_get_zero_initialized_remap();
+          if (RCL_RET_OK == _rcl_parse_remap_rule(argv[i + 1], allocator, rule)) {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "remap rule : %s\n", argv[i + 1]);
+            ++(args_impl->num_remap_rules);
+            i += 1;  // Skip flag here, for loop will skip rule.
+            continue;
+          }
+          RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
```

Comment:
Something else about error handling here (it applies to params too):
If the parsing of the remapping rule fails, we are not skipping it after that (`i += 1`), we're just skipping the flag and we will try to parse the "rule" as another thing later.
I don't know if that's good or not, we should probably improve error handling here (it can be done in a follow up).

Reply:
Yes, I know. It is as (in)correct as we deem it to be. We should probably fail entirely instead of moving forward though.